messaging/core_test_utils.go

```diff
@@ -27,27 +25,28 @@ func NewTestMessagingEnvironment() (*TestMessagingEnvironment, error) {
 	}
 
 	return &TestMessagingEnvironment{
-		waku: waku,
+		waku:  waku,
+		cores: make([]*Core, 0),
 	}, nil
 }
 
-func (f *TestMessagingEnvironment) Setup() error {
-	return f.waku.Start()
-}
-
 func (f *TestMessagingEnvironment) TearDown() error {
-	return f.waku.Stop()
+	for _, core := range f.cores {
+		if err := core.stop(); err != nil {
+			return err
+		}
+	}
+	f.cores = make([]*Core, 0)
+	return nil
 }
 
-func (f *TestMessagingEnvironment) NewTestCore(identity *ecdsa.PrivateKey, db *sql.DB, persistence types.Persistence, encryptionProtocol *encryption.Protocol, options ...Options) (*Core, error) {
-	return NewCore(
-		f.waku,
-		identity,
-		db,
-		persistence,
-		encryptionProtocol,
-		options...,
-	)
+func (f *TestMessagingEnvironment) NewTestCore(params CoreParams, options ...Options) (*Core, error) {
+	core, err := newCore(f.waku, params, newConfig(options...))
+	if err != nil {
+		return nil, err
+	}
+	f.cores = append(f.cores, core)
```

Comment:
Instead of this `f.cores`, maybe just use `t.Cleanup`?

```suggestion
	t.Cleanup(func() { core.stop() })
```

Reply:
Hm… good idea in general, except that `core_test_utils` is not aware of the testing package, and I would keep it that way since it is compiled together with the production code.